PolyEngine/Core/Src/Queue.hpp

```diff
@@ -96,42 +96,42 @@ namespace Poly
 		//------------------------------------------------------------------------------
 		void PushBack(const T& obj)
 		{
+			if (Empty() || Head == GetNextIdx(Tail))
```

Comment:
You can simply check if `GetSize() + 1 > GetCapacity()` here and in the other places.